### ISSUE TITLE
Inconsistent naming of tasks in GridScheduler

### ISSUE THREAD

**Issue by andrii0lomakin** [beehive-lab/TornadoVM#507]
Checking of the presence of WorkerGrid in GridScheduler is implemented inconsistently:
1. In uk.ac.manchester.tornado.runtime.tasks.TornadoTaskGraph#isTaskNamePresent check name of the task is concatinated with the name of the task graph, and then it is compared with a key stored in GridScheduler.

```
private boolean isTaskNamePresent(String taskName) {
        for (TaskPackage taskPackage : taskPackages) {
            if (taskName.equals(STR."\{taskGraphName}.\{taskPackage.getId()}")) {
                return true;
            }
        }
        return false;
    }
```
So, it is forcing the user to prepend the name of the task graph with the name of the task during the addition of WorkerGrid to the GridScheduler.
2. In method uk.ac.manchester.tornado.api.GridScheduler#contains the name of the task graph is added by GridScheduler itself, so if the user follows that first approach, this method always returns `false`. 

From my point of view, the first case is implemented incorrectly because, at the current stage, there is no validation of the correctness of the fact that the user provided the correct name to the GridScheduler, and there is no clear demand to do that in the GridScheduler contract.

So IMHO, both cases should be implemented over the call to GridScheduler#contains, and usage of the same scheduler for two task graphs at once should be prohibited (which is quite simple to implement).

**Comment by jjfumero** [comment]
If the `isTaskNamePresent` method returns `false`, you will get a Runtime Exception. I do not get why this is inconsistent. Could you please share a test case that breaks?

**Comment by andrii0lomakin** [comment]
@jjfumero yes you are right (I admit I missed this because encountered this issue long ago), but exception says `Grid scheduler with name \{gridName} not found in the Task-Graph` , while user clearly sees that it adds this task in task graph with the same name as in GridScheduler. IMHO that is confusing, I for example had to debug to understand root cause.
